packages/@aws-cdk/aws-appsync/lib/appsync-function.ts

```diff
@@ -128,11 +142,21 @@ export class AppsyncFunction extends Resource implements IAppsyncFunction {
 
   public constructor(scope: Construct, id: string, props: AppsyncFunctionProps) {
     super(scope, id);
+
+    // If runtime is specified, code must also be
+    if (props.runtime && !props.code) {
```

Comment:
mapping templates and code should also be mutually exclusive. can we add a check here as well?